vochain/scrutinizer/vote.go

```diff
@@ -167,78 +129,64 @@ func (s *Scrutinizer) addLiveResultsVote(envelope *models.Vote) error {
 		return err
 	}
 	if len(vote.Votes) > MaxQuestions {
-		return fmt.Errorf("too many questions on addVote")
-	}
-	processBytes, err := s.Storage.Get(s.Encode("liveProcess", pid))
-	if err != nil {
-		return fmt.Errorf("error adding vote to process %x, skipping addVote: (%s)",
-			pid, err)
-	}
-
-	var pv models.ProcessResult
-	if err := proto.Unmarshal(processBytes, &pv); err != nil {
-		return fmt.Errorf("cannot unmarshal vote (%s)", err)
-	}
-	if err := addVote(pv.Votes,
-		vote.Votes,
-		envelope.GetWeight(),
-		p.GetVoteOptions(),
-		p.GetEnvelopeType()); err != nil {
-		return err
-	}
-
-	processBytes, err = proto.Marshal(&pv)
-	if err != nil {
-		return err
-	}
-
-	if err := s.Storage.Put(s.Encode("liveProcess", pid), processBytes); err != nil {
-		return err
+		return fmt.Errorf("too many elements on addVote")
 	}
 
-	log.Debugf("addVote %v on process %x", vote.Votes, pid)
-	return nil
+	return s.db.UpdateMatching(&Results{},
+		badgerhold.Where("ProcessID").Eq(pid),
+		func(record interface{}) error {
```

Comment:
would be nice if badgerhold allowed you to use `func(*Results) error` here, just like with ForEach. Maybe we could file an issue about that.

Reply:
Sure